### PR TITLE
Closes #37 - Improve logging in LdapClient

### LINKED ISSUE #37: Improve logging in LdapClient
As of now there is only a very basic logging in LdapClient. This makes the debugging of problems that come from LDAP very difficult. The goal should be to log enough information to reconstruct and debug each step of a communication with a LDAP server, without having to know the specifics/configuration of the server in question.

### PR DESCRIPTION
**Pull request by CRoberto1926**
### Thanks for your PR! Please fill out the following list :)

---

- [x] I put the ticket or multiple tickets in review
- [x] Commit message format → Closes #&lt;Issue Number&gt; - Your commit message.
- [x] Sonarcloud link : https://sonarcloud.io/summary/new_code?id=CRoberto1926_kadai&branch=feature%2Fissue-37-ldap-client-logging
- [x] No documentation update needed
- [ ] Link to PR with documentation update: \<add the link here>
- [x] No Release Notes needed
- [ ] Release Notes :

```

```

### CODE REVIEW

File: common/kadai-common/src/main/java/io/kadai/common/internal/util/LogSanitizer.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 package io.kadai.common.internal.util;
 
+// TODO FIXME replace with OWASP SANITIZER: https://github.com/OWASP/owasp-java-encoder
```
Comment (holgerhagen) [inline common/kadai-common/src/main/java/io/kadai/common/internal/util/LogSanitizer.java]: I suggest we rather open an issue than create a TODO comment.

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java [hunk 2]
```diff
@@ -840,6 +848,179 @@ void testMinSearchForLength(final String name) throws InvalidArgumentException {
     }
   }
 
+  private AccessIdRepresentationModel lookupAndDebugLog(
```
Comment (holgerhagen) [inline rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]: The fact that we write debug logs should not be in the operation name, shouldn't it? It is rather a side effect depending on the log level.

**Review by holgerhagen — CHANGES_REQUESTED** [review]
Please add guarded logging to all statements (if isDebugEnabled...)

File: common/kadai-common/src/main/java/io/kadai/common/internal/util/LogSanitizer.java (hunk 1 above)
Comment (CRoberto1926) [reply common/kadai-common/src/main/java/io/kadai/common/internal/util/LogSanitizer.java]: Done: https://github.com/kadai-io/kadai/issues/69

File: rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java (hunk 2 above)
Comment (CRoberto1926) [reply rest/kadai-rest-spring/src/main/java/io/kadai/common/rest/ldap/LdapClient.java]: I prefer to be as precise as possible in a method name. In this was the reader doesn't have to read the method definition. But you also have a point, so I removed "AndDebugLog" from the "search" and "lookup" methods.

**Pull request closed without merging**